packages/react/src/App.ts

```diff
@@ -25,17 +26,25 @@ export default function App({
     )
   }, [])
 
+  useLayoutEffect(() => {
+    if (!current.preserveScroll) {
+      router.resetScrollPositions()
+    }
+  }, [current])
+
   useEffect(() => {
     router.init({
       initialPage,
       resolveComponent,
-      swapComponent: async ({ component, page, preserveState }) => {
+      swapComponent: async ({ component, page, preserveState, preserveScroll }) => {
         setCurrent((current) => ({
           component,
           page,
           key: preserveState ? current.key : Date.now(),
+          preserveScroll: !!preserveScroll,
```

Comment:
both `preserveState` and `preserveScroll` are typed as a `PreserveStateOption` in the router core, although they actually seem to be resolved to booleans where it matters in the usage here?

This is where my knowledge of this codebase starts to feel super minimal -- so I may be way off here